src/uct/ib/rc/accel/rc_mlx5_ep.c

```diff
@@ -579,6 +579,11 @@ ucs_status_t uct_rc_mlx5_ep_fc_ctrl(uct_ep_t *tl_ep, unsigned op,
      * messages are bundled with AM. */
     ucs_assert(op == UCT_RC_EP_FC_PURE_GRANT);
 
+    if (ucs_unlikely(ep->tx.wq.super.verbs.qp->state != IBV_QPS_RTS)) {
+        return (ep->tx.wq.super.verbs.qp->state == IBV_QPS_INIT) ? UCS_OK :
```

Comment:
in master branch, we'll need to use rc ep flags

Reply:
to avoid extra pointer dereference or by any other reason?
@hoopoepg FYI